### ISSUE TITLE
[ENH]: Alternative Dependency `soxr`

### ISSUE THREAD

**Issue by robertgshaw2-neuralmagic** [librosa/librosa#1869]
### Problem

Hello, I am a committer on the `vllm-project`. 

We recently added `librosa` as a dependency for `vllm` as we work to support audio models. A user of the `vllm-project` for a large institution flagged that `librosa` has a dependency on [`soxr`](https://pypi.org/project/soxr/), which appears to be an LGPL license and is preventing usage in that institution:
- https://github.com/vllm-project/vllm/issues/8030

We are going to move `librosa` as an optional dependency in `vllm` for now to alleviate the concerns

I understand that LGPL is designed for this exact case (i.e. that you can still use it as a library in proprietary/non-copyleft software), but nonetheless we want the `vllm-project` to be as "pure" as possible licensing-wise especially for optional features

### Proposed solution

I am not sure how integral `soxr` is to the `librosa` codebase, but is there any way that this could be moved to an optional dependency or replaced with a more permissive alternative, it would help a lot to alleviate issues with usage of `librosa` in downstream applications

**Comment by bmcfee** [comment]
It's a tricky question.  To cut to the chase, my inclination here is to leave things as is - as you say, this is exactly what LGPL is for.  At the risk of sounding glib, if large commercial actors are touchy about this, that seems like their problem to work around.  As a developer, I totally understand where you're coming from with trying to keep licensing as permissive as possible, but I'm also sympathetic to the soxr devs' position.

I'm much less sympathetic to said large commercial actors, who if they care that much about it, should just pony up and develop their own tools with an acceptable license.  In fact, the resampy package only exists because I needed a permissively licensed resampler and libsamplerate was GPL at the time, so I made my own.  Things are different now, thankfully.

More to the point here though: soxr isn't exactly integral to librosa.  The situation is as follows:

- We support multiple sample rate conversion backends: soxr, libsamplerate, resampy, and scipy.  
- Among these options, we had to choose a default, and soxr is both the fastest and highest quality.
- Technically, we could have made all of them optional dependencies, but AFAIK there's no way in a python packaging spec to indicate that "at least one of {x,y,z}" must be installed.  Hence the strict dependency.

The easiest workaround here is to install soxr but not use it; this is easily accomplished by overriding the `res_type` parameter at relevant call sites.  Any of the `sinc_*` values will use libsamplerate and will be of high quality, just a little slower than soxr.

If it's not acceptable to have soxr installed at all, then you're in the territory of hacking dependencies, and I'm not sure I can offer much here.  One option would be to specify that librosa is a dependency but should be installed with --no-deps (and then clone all other dependencies into your own package spec); I think this might be possible.  The other option is to have a post-install step that removes soxr from your env after everything is setup; both cases will leave you in a slightly inconsistent package state, but it will work just fine.

---

Longer term, I'm not opposed to implementing an install flag on our side that would skip soxr in favor libsamplerate if people are really itching for it.  However, we'd need to work out a clean way to handle the default parameter values in the affected code paths, and that seems like a dicey proposition.  But if someone wants to propose a mechanism for this, I'd be interested in thinking it through.
